custom-standards/Flyeralarm/Sniffs/Docblock/ReturnTypeSniff.php

```diff
@@ -145,4 +136,55 @@ private function isStartingWithUppercaseLetter($singleCharacter)
 
         return false;
     }
+
+    /**
+     * @param string $subject The return type string to check.
+     * @return void
+     */
+    private function checkReturnTypeShape(string $subject)
+    {
+        preg_match_all('#(?<separator>\s*\|\s*)?(?<atom>[^<>\|]+)(?<generic><(?<nested>.*)>)?#', $subject, $matches);
+
+        if (implode('', $matches[0]) !== $subject) {
```

Comment:
There is the tiny case of `$subject = ''` that currently is not covered.
Adding a check for the return value of `preg_match_all` being false should catch that.

```suggestion
        $matched = preg_match_all('#(?<separator>\s*\|\s*)?(?<atom>[^<>\|]+)(?<generic><(?<nested>.*)>)?#', $subject, $matches);

        if ($matched === false || implode('', $matches[0]) !== $subject) {
```

Reply:
Wouldn't that just cover errors in the RegEx? For an empty string, `preg_match_all` should return 0 since it doesn't match anything. We probably want to cover both cases:

```suggestion
        $matched = preg_match_all('#(?<separator>\s*\|\s*)?(?<atom>[^<>\|]+)(?<generic><(?<nested>.*)>)?#', $subject, $matches);

        if (!$matched || implode('', $matches[0]) !== $subject) {
```

or to be more explicit, because above it might not be obvious that it's intentionally checking for 0 or false:

```suggestion
        $matched = preg_match_all('#(?<separator>\s*\|\s*)?(?<atom>[^<>\|]+)(?<generic><(?<nested>.*)>)?#', $subject, $matches);

        if ($matched === 0 || $matched === false || implode('', $matches[0]) !== $subject) {
```

Reply:
For readability reasons, I prefer the first one.